Assets/Scripts/Controllers/Events/TradeController.cs

```diff
@@ -54,18 +54,12 @@ public void CallTradeShipTest(Furniture landingPad)
         go.transform.parent = WorldController.Instance.transform;
         TraderShipController controller = go.AddComponent<TraderShipController>();
         TradeShips.Add(controller);
-        controller.Trader = trader;
-        controller.Speed = 5f;
         go.transform.position = new Vector3(-10, 50, 0);
-        controller.LandingCoordinates = new Vector3(landingPad.Tile.X + 1, landingPad.Tile.Y + 1, 0);
-        controller.LeavingCoordinates = new Vector3(100, 50, 0);
         go.transform.localScale = new Vector3(1, 1, 1);
         SpriteRenderer spriteRenderer = go.AddComponent<SpriteRenderer>();
         spriteRenderer.sprite = SpriteManager.GetSprite("Trader", prototype.AnimationIdle.CurrentFrameName);
         spriteRenderer.sortingLayerName = "TradeShip";
-        controller.AnimationFlying = prototype.AnimationFlying.Clone();
-        controller.AnimationIdle = prototype.AnimationIdle.Clone();
-        controller.Renderer = spriteRenderer;
+        controller.Init(leavingCoords: new Vector3(100, 50, 0), landingCoords: new Vector3(landingPad.Tile.X + 1, landingPad.Tile.Y + 1, 0), speed: 5f, trader: trader, animationIdle: prototype.AnimationIdle.Clone(), animationFlying: prototype.AnimationFlying.Clone(), renderer: spriteRenderer);
```

Comment:
Yep, I agree.  I'll place one in.  If only constructors worked on Monobehaviours... I just made it as 'simple' since I didn't want to branch out too much (out of scope).